go/arrow/compute/datumkind_string.go

```diff
@@ -0,0 +1,29 @@
+// Code generated by "stringer -type=DatumKind -linecomment"; DO NOT EDIT.
+
+package compute
+
+import "strconv"
+
+func _() {
+	// An "invalid array index" compiler error signifies that the constant values have changed.
+	// Re-run the stringer command to generate them again.
+	var x [1]struct{}
+	_ = x[KindNone-0]
+	_ = x[KindScalar-1]
+	_ = x[KindArray-2]
+	_ = x[KindChunked-3]
+	_ = x[KindRecord-4]
+	_ = x[KindTable-5]
+	_ = x[KindCollection-6]
+}
+
+const _DatumKind_name = "nonescalararraychunked_arrayrecord_batchtablecollection"
+
+var _DatumKind_index = [...]uint8{0, 4, 10, 15, 28, 40, 45, 55}
```

Comment:
Not sure I understand this array?

Reply:
this file is actually a generated file using the "stringer" command, amusingly if you look at this and the value of `_DatumKind_name` together, they are effectively an arrow string array. these are the offsets into the string to get the `.String()` value for the enum values.

But like i said, this is generated by running `go generate` and is not something hand written.